lms/static/scripts/frontend_apps/components/YouTubePicker.tsx

```diff
@@ -19,28 +22,35 @@ export default function YouTubePicker({
   onSelectURL,
 }: YouTubePickerProps) {
   const inputRef = useRef<HTMLInputElement | null>(null);
-  const [videoId, setVideoId] = useState(
-    (defaultURL && videoIdFromYouTubeURL(defaultURL)) || null
+  const [currentURL, setCurrentURL] = useState(defaultURL);
+  const videoId = useMemo(
+    () => (currentURL ? videoIdFromYouTubeURL(currentURL) : null),
+    [currentURL]
   );
-  const [error, setError] = useState<string>();
-  const resetToInitialState = () => {
-    setVideoId(null);
-    setError(undefined);
-  };
-  const { thumbnail, metadata } = useYouTubeVideoInfo(videoId);
+  const videoInfo = useAPIFetch<YouTubeVideoInfo>(
+    videoId ? urlPath`/api/youtube/videos/${videoId}` : null
+  );
+  const error = useMemo(() => {
+    const defaultError =
+      'URL must be a YouTube video, e.g. "https://www.youtube.com/watch?v=cKxqzvzlnKU"';
 
-  const verifyURL = (inputURL: string) => {
-    const videoId = videoIdFromYouTubeURL(inputURL);
-    if (videoId) {
-      setVideoId(videoId);
-      setError(undefined);
-    } else {
-      setVideoId(null);
-      setError(
-        'URL must be a YouTube video, e.g. "https://www.youtube.com/watch?v=cKxqzvzlnKU"'
-      );
+    // We have a fetch error. A valid YouTube URL was set, but the API could not find the video
+    if (videoInfo.error && isAPIError(videoInfo.error)) {
+      return videoInfo.error.errorCode === 'youtube_video_not_found'
+        ? 'Video not found'
+        : defaultError;
     }
-  };
+
+    // A URL was set, but it's not a valid YouTube URL
+    if (currentURL && !videoId) {
+      return defaultError;
+    }
+
+    return undefined;
+  }, [currentURL, videoId, videoInfo.error]);
+
+  const onURLChange = (inputURL: string) => setCurrentURL(inputURL);
+  const resetCurrentUrl = () => setCurrentURL(undefined);
```

Comment:
```suggestion
  const resetCurrentURL = () => setCurrentURL(undefined);
```

Reply:
Ugh, I keep forgetting about this 😅 

I'm starting to think we need a linting rule, haha